test(Configuration Rubric Service)

lomamech opened the pull request:
- Implememt Unit Test for Configuration Rubric Service
- Remove unused service $uibModal

closes #3083

lomamech committed:
test(Configuration Rubric Service)
- Implememt Unit Test for Configuration Rubric Service
- Remove unused service $uibModal

jniles commented on test/client-unit/services/rubricsConfigurationService.spec.js:
```diff
@@ -0,0 +1,74 @@
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('Count the Numbers of Rubrics Configured For the period 1', () => {
```
This looks good!

One thing that might be helpful is comparing the different between `RubricsConfigs.getRubrics()` and `RubricsConfigs.getRubrics(1)`.  These should return two different results, I think.

jniles reviewed (approved):
@lomamech good first client unit test!

In general, it isn't that big a deal to test HTTP requests, since most of the logic is either on the server or executed before calling the HTTP request.  That said, this demonstrates a good understanding of how to set up a test!

If you want a real challenge, check out https://github.com/IMA-WorldHealth/bhima-2.X/issues/2650 - writing tests for the PurchaseFormService!

lomamech committed:
test(Configuration Rubrics Service)
- Adding unit test for Throw Error Where Trying to Rubrics Configured for Undefined Period Id

jniles commented on client/src/modules/payroll/rubric_configuration/configuration.service.js:
```diff
@@ -10,14 +10,20 @@ ConfigurationService.$inject = ['PrototypeApiService', '$uibModal', '$http', 'ut
 
   // loads the configuration's rubrics
   function getRubrics(id) {
+    if (angular.isUndefined(id)) {
```
Nice 👍

jniles commented on test/client-unit/services/rubricsConfigurationService.spec.js:
```diff
@@ -0,0 +1,79 @@
+    expect(value).to.have.length(4);
+  });
+
+  it('Count the Numbers of Rubrics Configured For the period Undefined', () => {
```
A better description should probably tell you exactly what is happening: "getRubrics() will throw if the period is undefined".

jniles commented on test/client-unit/services/rubricsConfigurationService.spec.js:
```diff
@@ -0,0 +1,79 @@
+  });
+
+  it('Count the Numbers of Rubrics Configured For the period Undefined', () => {
+    let value;
```
You can remove this `let value` - it isn't used from what I can tell.

jniles reviewed (approved):
Looks good to me!  Well done @lomamech!

bors r+
